R/build-home-index.R

```diff
@@ -39,17 +45,50 @@ data_home <- function(pkg = ".") {
 
 data_home_sidebar <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
-  if (!is.null(pkg$meta$home$sidebar))
+  if (isFALSE(pkg$meta$home$sidebar))
     return(pkg$meta$home$sidebar)
 
+  sidebar_path <- file.path(pkg$src_path, pkg$meta$home$sidebar$html)
+
+  if (length(sidebar_path)) {
+
+    if (!file.exists(sidebar_path)) {
+      abort(sprintf("Can't find file %s.", pkg$meta$home$sidebar$html))
+    }
+    return(paste0(read_lines(sidebar_path), collapse = "\n"))
+  }
+
+  if (is.null(pkg$meta$home$sidebar$structure)) {
+    return(paste0(
```

Comment:
I think you should make this into a named list, and then reorganise based on `sidebar$structure`

Reply:
Now done, but is it bad that one computes e.g. the authors section even if it ends up not being used?

Reply:
I don't think it's a big deal. If it later turns out to be a pain, we can figure out some mechanism to delay evaluation until it's really needed.